src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -201,15 +201,12 @@ public IPAddress(ReadOnlySpan<byte> address)
         private static ushort[] ReadUInt16NumbersFromBytes(ReadOnlySpan<byte> address)
         {
             ushort[] numbers = new ushort[NumberOfLabels];
-            if (Vector128.IsHardwareAccelerated)
+            if (Vector128.IsHardwareAccelerated && BitConverter.IsLittleEndian)
             {
-                Vector128<ushort> ushorts = Vector128.LoadUnsafe(ref MemoryMarshal.GetReference(address)).AsUInt16();
-                if (BitConverter.IsLittleEndian)
-                {
-                    // Reverse endianness of each ushort
-                    ushorts = Vector128.ShiftLeft(ushorts, 8) | Vector128.ShiftRightLogical(ushorts, 8);
-                }
-                ushorts.StoreUnsafe(ref MemoryMarshal.GetArrayDataReference(numbers));
+                Vector128<ushort> ushorts = Vector128.Create(address).AsUInt16();
+                // Reverse endianness of each ushort
+                ushorts = (ushorts << 8) | (ushorts >> 8);
+                ushorts.StoreUnsafe(ref numbers[0]);
             }
             else
             {
```

Comment:
Could we replace this with `if (BitConverter.IsLittleEndian) BinaryPrimitives.ReverseEndianness(numbers, numbers)`?

Reply:
> Could we replace this with `if (BitConverter.IsLittleEndian) BinaryPrimitives.ReverseEndianness(numbers, numbers)`?

hm.. good point

Reply:
Unfortunately jit can't properly optimize it, `ReverseEndianness` for spans is pretty big